osu.Game.Rulesets.Osu/Edit/OsuHitObjectComposer.cs

```diff
@@ -112,6 +112,14 @@ public override string ConvertSelectionToString()
 
         private RectangularPositionSnapGrid rectangularPositionSnapGrid;
 
+        protected override double ReadCurrentDistanceSnap(HitObject before, HitObject after)
```

Comment:
Spinners throw a bit of a wrench into this, the function proposes distance spacings for spinners that are a bit nonsense (likely because spinners technically have a "position" in the middle of the playfield). May want to prevent proposing a spacing if one of the objects is a spinner.

Reply:
I think I'd probably leave this for now as "working as intended", based on the gameplay direction where we want to allow non-centered spinners.